[CI] ExchangeServiceTests testFailToRespondPage failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/xqf2tdt4qwvsm/tests/:x-pack:plugin:esql:compute:test/org.elasticsearch.compute.operator.exchange.ExchangeServiceTests/testFailToRespondPage

**Reproduction line:**
```
./gradlew ':x-pack:plugin:esql:compute:test' --tests "org.elasticsearch.compute.operator.exchange.ExchangeServiceTests.testFailToRespondPage" -Dtests.seed=CB4F0EDC716ADF9C -Dtests.locale=tr -Dtests.timezone=Europe/Busingen -Druntime.java=20
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.compute.operator.exchange.ExchangeServiceTests&tests.test=testFailToRespondPage

**Failure excerpt:**
```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=1287, name=elasticsearch[test][generic][T#3], state=RUNNABLE, group=TGRP-ExchangeServiceTests]


  Caused by: java.lang.AssertionError: TaskCancellationService is not initialized

    at __randomizedtesting.SeedInfo.seed([CB4F0EDC716ADF9C]:0)
    at org.elasticsearch.tasks.TaskManager.getCancellationService(TaskManager.java:261)
    at org.elasticsearch.tasks.TaskManager.cancelTaskAndDescendants(TaskManager.java:819)
    at org.elasticsearch.tasks.TaskManager$4.doRun(TaskManager.java:806)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983)
    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
    at java.lang.Thread.run(Thread.java:1623)

```

--- a/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/exchange/ExchangeServiceTests.java
+++ b/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/exchange/ExchangeServiceTests.java
@@ -7,7 +7,6 @@
 
 package org.elasticsearch.compute.operator.exchange;
 
-import org.apache.lucene.tests.util.LuceneTestCase;
 import org.elasticsearch.ExceptionsHelper;
 import org.elasticsearch.TransportVersion;
 import org.elasticsearch.action.ActionListener;
@@ -30,6 +29,7 @@ import org.elasticsearch.compute.operator.SinkOperator;
 import org.elasticsearch.compute.operator.SourceOperator;
 import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskCancellationService;
 import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.test.transport.MockTransportService;
 import org.elasticsearch.test.transport.StubbableTransport;
@@ -59,7 +59,6 @@ import java.util.stream.IntStream;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasSize;
 
-@LuceneTestCase.AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/98707")
 public class ExchangeServiceTests extends ESTestCase {
 
     private TestThreadPool threadPool;
@@ -415,6 +414,7 @@ public class ExchangeServiceTests extends ESTestCase {
             null,
             Collections.emptySet()
         );
+        service.getTaskManager().setTaskCancellationService(new TaskCancellationService(service));
         service.start();
         service.acceptIncomingRequests();
         return service;

